Bugs in the new improved pattern matching

**Version Used**: 64c74c0ac3ccf45d63ae14d7d2ef067cb21b75ad

**Steps to Reproduce**:
```csharp
var x = 5;
var str = x switch
{
    1 => "a",
    > 2 => "b",
    > 1 and <= 2 => "c",
    < 0 => "d"
};
```
See: https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQAdYwggdsgZgBoYQpkBbAHwAEAmARgFgAoGggAno4GEOBvVh2FcGANi4AWDgFkoAS1wAKAJQChIzQDcoCDgA8OAXg4BWANwbNwnXuQw9Jw8gDu8mAGMAFleuCW1oEcDMYAfBwARFARRL5BHOF0YZHAMXFB4SFQuAAmHAA8JklG4REeaQHxIvkcAAzJETkR6cIAvpaVgTQMAJxK9ggqHZqtrK1AA=

**Expected Behavior**: 
Compiled successfully

**Actual Behavior**:
error CS8510: The pattern has already been handled by a previous arm of the switch expression.

However, if I change my code to
```csharp
var x = 5;
var str = x switch
{
    1 => "a",
    > 2 => "b",
    <= 2 and > 1 => "c",
    < 0 => "d"
};
```
It will work as expect.

I will add a test to demonstrate that there is no bug in recent versions of the pattern-matching branch:

```csharp
        [Fact, WorkItem(43308, "https://github.com/dotnet/roslyn/issues/43308")]
        public void RelationalEdgeTest_01()
        {
            var source =
@"using System;
class C
{
    static void Main()
    {
        var x = 5;
        var str = x switch // does not handle zero
        {
            1 => ""a"",
            > 2 => ""b"",
            > 1 and <= 2 => ""c"",
            < 0 => ""d""
        };
        Console.Write(str);
        str = x switch // does not handle zero
        {
            1 => ""a"",
            > 2 => ""b"",
            <= 2 and > 1 => ""c"",
            < 0 => ""d""
        };
        Console.Write(str);
    }
}";
            string expectedOutput = "bb";
            var compilation = CreateCompilation(source, options: TestOptions.ReleaseExe, parseOptions: TestOptions.RegularWithPatternCombinators);
            compilation.VerifyDiagnostics(
                // (7,21): warning CS8509: The switch expression does not handle all possible values of its input type (it is not exhaustive).
                //         var str = x switch // does not handle zero
                Diagnostic(ErrorCode.WRN_SwitchExpressionNotExhaustive, "switch").WithLocation(7, 21),
                // (15,17): warning CS8509: The switch expression does not handle all possible values of its input type (it is not exhaustive).
                //         str = x switch // does not handle zero
                Diagnostic(ErrorCode.WRN_SwitchExpressionNotExhaustive, "switch").WithLocation(15, 17)
                );
            var compVerifier = CompileAndVerify(compilation, expectedOutput: expectedOutput);
        }
```

--- a/src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests3.cs
+++ b/src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests3.cs
@@ -4681,5 +4681,46 @@ static class Assert
                 _ => throw new System.InvalidOperationException(),
             });
         }
+
+        [Fact, WorkItem(43308, "https://github.com/dotnet/roslyn/issues/43308")]
+        public void RelationalEdgeTest_01()
+        {
+            var source =
+@"using System;
+class C
+{
+    static void Main()
+    {
+        var x = 5;
+        var str = x switch // does not handle zero
+        {
+            1 => ""a"",
+            > 2 => ""b"",
+            > 1 and <= 2 => ""c"",
+            < 0 => ""d""
+        };
+        Console.Write(str);
+        str = x switch // does not handle zero
+        {
+            1 => ""a"",
+            > 2 => ""b"",
+            <= 2 and > 1 => ""c"",
+            < 0 => ""d""
+        };
+        Console.Write(str);
+    }
+}";
+            string expectedOutput = "bb";
+            var compilation = CreateCompilation(source, options: TestOptions.ReleaseExe, parseOptions: TestOptions.RegularWithPatternCombinators);
+            compilation.VerifyDiagnostics(
+                // (7,21): warning CS8509: The switch expression does not handle all possible values of its input type (it is not exhaustive).
+                //         var str = x switch // does not handle zero
+                Diagnostic(ErrorCode.WRN_SwitchExpressionNotExhaustive, "switch").WithLocation(7, 21),
+                // (15,17): warning CS8509: The switch expression does not handle all possible values of its input type (it is not exhaustive).
+                //         str = x switch // does not handle zero
+                Diagnostic(ErrorCode.WRN_SwitchExpressionNotExhaustive, "switch").WithLocation(15, 17)
+                );
+            var compVerifier = CompileAndVerify(compilation, expectedOutput: expectedOutput);
+        }
     }
 }

